src/components/DateInput.jsx

```diff
@@ -159,6 +162,8 @@ class DateInput extends React.PureComponent {
     } else if (key === '?') {
       e.preventDefault();
       onKeyDownQuestionMark(e);
+    } else if (key === 'Escape') {
```

Comment:
Seems like you'd also want to add escape key handling within the calendar overlay itself as well.